aten/src/ATen/native/mps/operations/ConstantOps.mm

```diff
@@ -79,16 +63,26 @@
 }
 
 Tensor& zero_mps_(Tensor& self) {
-  return at::native::fill_scalar_mps_impl(self, 0.0f);
+  if (self.is_contiguous()) {
+    MPSStream* stream = getCurrentMPSStream();
+    auto storage_byte_offset = self.storage_offset() * self.itemsize();
+    stream->fill(mps::getMTLBufferStorage(self), 0, self.nbytes(), storage_byte_offset);
```

Comment:
Do we need to use the updated logic of nbytes @DenisVieriu97 here?

Reply:
Can you try running this test to see if it crashes in cases where the storage_offset gets bigger than self.nbytes?
```
def test_fill_zeros(self):
        n_tensors= 100
        n_tensor_elems = 784
        elems_cpu = torch.arange(n_tensors * n_tensor_elems, dtype=torch.float32)
        elems_mps = torch.arange(n_tensors * n_tensor_elems, dtype=torch.float32, device="mps")

        tensor_list_cpu = []
        tensor_list_mps = []
        for i in range(0, n_tensors - 1):
            # create a list of contiguous view tensors (view tensor created by the slice op)
            t_cpu = elems_cpu[n_tensor_elems * i : n_tensor_elems * (i + 1)]
            t_mps = elems_mps[n_tensor_elems * i : n_tensor_elems * (i + 1)]

            tensor_list_cpu.append(t_cpu)
            tensor_list_mps.append(t_mps)

        for i in range(0, n_tensors - 1):
            zeros_cpu = tensor_list_cpu[i].view(1, 784).zero_()
            zeros_mps = tensor_list_mps[i].view(1, 784).zero_()

            self.assertEqual(zeros_cpu, zeros_mps.cpu())
```

Reply:
Like we discussed, the test passes and I'll keep the is_contiguous() only.